File: testdata/Sample.java
package com.example.sample;

import java.util.List;
import java.util.ArrayList;
import java.io.Serializable;

/**
 * Represents a generic repository for storing entities.
 *
 * @param <T> the type of entity stored
 */
public interface Repository<T> {

    /**
     * Find an entity by its ID.
     *
     * @param id the entity ID
     * @return the entity, or null if not found
     */
    T findById(long id);

    /**
     * Save an entity.
     *
     * @param entity the entity to save
     * @return the saved entity
     */
    T save(T entity);

    /**
     * Delete an entity by its ID.
     *
     * @param id the entity ID
     */
    void deleteById(long id);

    /**
     * Return all entities.
     *
     * @return list of all entities
     */
    List<T> findAll();
}

/**
 * Status of a user account.
 */
enum AccountStatus {
    ACTIVE,
    INACTIVE,
    SUSPENDED,
    DELETED
}

/**
 * Represents a user in the system.
 *
 * Implements Repository for self-management and Serializable for persistence.
 */
@SuppressWarnings("unchecked")
public class User implements Repository<User>, Serializable {

    private static final long serialVersionUID = 1L;

    private long id;
    private String name;
    private String email;
    private AccountStatus status;

    /**
     * Create a new User with the given name and email.
     *
     * @param name  the user's name
     * @param email the user's email address
     */
    public User(String name, String email) {
        this.name = name;
        this.email = email;
        this.status = AccountStatus.ACTIVE;
    }

    @Override
    public User findById(long id) {
        return null;
    }

    @Override
    public User save(User entity) {
        return entity;
    }

    @Override
    public void deleteById(long id) {
        // no-op
    }

    @Override
    public List<User> findAll() {
        return new ArrayList<>();
    }

    /**
     * Get the user's display name.
     *
     * @return formatted display name
     */
    public String getDisplayName() {
        return name + " <" + email + ">";
    }

    @Deprecated
    public String getLegacyId() {
        return String.valueOf(id);
    }

    /**
     * Check if the user account is active.
     *
     * @return true if active
     */
    public boolean isActive() {
        return status == AccountStatus.ACTIVE;
    }

    @Override
    public String toString() {
        return "User{name='" + name + "', email='" + email + "'}";
    }
}

File: internal/parser/csharp/testdata/sample.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace MyApp.Models
{
    /// <summary>
    /// Interface for repository operations.
    /// </summary>
    public interface IRepository<T>
    {
        T FindById(int id);
        IEnumerable<T> GetAll();
        void Save(T entity);
        void Delete(int id);
    }

    /// <summary>
    /// Represents a user in the system.
    /// </summary>
    public class User
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public bool IsActive { get; set; }

        public User(int id, string name, string email)
        {
            Id = id;
            Name = name;
            Email = email;
            IsActive = true;
        }

        /// <summary>
        /// Gets the display name.
        /// </summary>
        public string GetDisplayName()
        {
            return $"{Name} ({Email})";
        }

        public bool Validate()
        {
            return !string.IsNullOrEmpty(Name) && !string.IsNullOrEmpty(Email);
        }
    }

    /// <summary>
    /// Account status values.
    /// </summary>
    public enum AccountStatus
    {
        Active,
        Inactive,
        Suspended,
        Deleted
    }

    /// <summary>
    /// A point in 2D space.
    /// </summary>
    public struct Point
    {
        public double X { get; set; }
        public double Y { get; set; }

        public double DistanceTo(Point other)
        {
            return Math.Sqrt(Math.Pow(X - other.X, 2) + Math.Pow(Y - other.Y, 2));
        }
    }

    /// <summary>
    /// User repository implementation.
    /// </summary>
    public class UserRepository : IRepository<User>
    {
        private readonly List<User> _users = new List<User>();
        public const int MaxUsers = 1000;

        public User FindById(int id)
        {
            return _users.FirstOrDefault(u => u.Id == id);
        }

        public IEnumerable<User> GetAll()
        {
            return _users.AsReadOnly();
        }

        public void Save(User entity)
        {
            if (!entity.Validate())
            {
                throw new ArgumentException("Invalid user");
            }
            _users.Add(entity);
        }

        public void Delete(int id)
        {
            var user = FindById(id);
            if (user != null)
            {
                _users.Remove(user);
            }
        }
    }
}
